### PR TITLE
Add formio checkbox and refactor privacy checkbox component

### PR DESCRIPTION
**Pull request by sergei-maertens**
Prerequisite for open-formulieren/open-forms#3203

This refactors the privacy checkbox to use Formik & formio rendering inside formik form.

### CODE REVIEW

File: src/components/Summary/CosignSummary.js [hunk 1]
```diff
@@ -14,25 +13,22 @@ import useSessionTimeout from 'hooks/useSessionTimeout';
...
   onDataLoaded,
   onCosignComplete,
   onDestroySession,
-  onPrivacyCheckboxChange,
 }) => {
   const intl = useIntl();
-  const config = useContext(ConfigContext);
 
   // The backend has added the submission to the session, but we need to load it
   // eslint-disable-next-line
```
Comment (SilviaAmAm) [inline src/components/Summary/CosignSummary.js:30]: This maybe can be removed now that the unused variable has been removed?

File: src/components/Summary/CosignSummary.js [hunk 2]
```diff
@@ -89,7 +76,7 @@ const CosignSummary = ({
   };
 
   // eslint-disable-next-line
```
Comment (SilviaAmAm) [inline src/components/Summary/CosignSummary.js:78]: Same remark as above :upside_down_face:

File: src/components/Summary/SubmissionSummary.js [hunk 3]
```diff
@@ -67,25 +54,17 @@ const SubmissionSummary = ({
...
+    return await loadSummaryData(submissionUrl);
   }, [refreshedSubmission.url]);
 
   if (error) {
     console.error(error);
   }
 
-  const onSubmit = async event => {
-    event.preventDefault();
+  const onSubmit = async ({privacy: privacyPolicyAccepted}) => {
```
Comment (SilviaAmAm) [inline src/components/Summary/SubmissionSummary.js:64]: I don't seem to be able to submit a form anymore :thinking: It seems that here the `onSubmit` function still receives an event?

**Review by SilviaAmAm — CHANGES_REQUESTED** [review]

File: src/components/Summary/SubmissionSummary.js (hunk 3 above)
Comment (sergei-maertens) [reply src/components/Summary/SubmissionSummary.js:64]: shit yeah I knew that I still had to test end-to-end!

**Review by bart-maykin — APPROVED** [review]

**Review by SilviaAmAm — APPROVED** [review]

**Pull request merged**